lib/find-node-directory.js

```diff
@@ -0,0 +1,40 @@
+var path = require('path')
+  , log = require('npmlog')
+
+function findNodeDirectory() {
+  // Have a look to see what is above us, to try and work out where we are
+  npm_parent_directory = path.join(__dirname, '../../../..')
+  log.verbose('node-gyp root', 'npm_parent_directory is ' + path.basename(npm_parent_directory))
+  node_root_dir = ""
+
+  log.verbose('node-gyp root', 'Finding node root directory')
+  if (path.basename(npm_parent_directory) == 'lib') {
+    // We are in a node install directory
```

Comment:
From the history I can gather the function to get the Node directory used to exist somewhere else as well but no longer does.  Looking at the potential installation configurations I agree that 'lib' does not make sense.  The cases I've been testing include when you build from source and have that layout and the case were you install from the tar.gz.  From those I think we should be checking 2 cases. One where the parent is "node_modules" (which is covered) and one where the parent is "deps". I'll update and add more comments to try to make it clear which cases are being covered.